### PR TITLE
Fix handling of styles

### PR DESCRIPTION
**Pull request by dontub**
Fixes issues reported in https://github.com/systopia/de.systopia.civioffice/pull/62#pullrequestreview-2101152138.

systopia-reference: 25379

### CODE REVIEW

File: tests/phpunit/bootstrap.php [hunk 1]
```diff
@@ -18,6 +18,12 @@
 
 $loader->register();
 
+// Make CRM_Civioffice_ExtensionUtil available.
+require_once __DIR__ . '/../../civioffice.civix.php';
+
+// Ensure function ts() is available - it's declared in the same file as CRM_Core_I18n.
+\CRM_Core_I18n::singleton();
```
Comment (jensschuppe) [inline tests/phpunit/bootstrap.php:25]: This is not the case anymore with CiviCRM `5.74`+, see civicrm/civicrm-core@b09cee8ecac6108801aa697f8c921421d92de0bf

File: CRM/Civioffice/DocumentRendererType/LocalUnoconv/PhpWordTemplateProcessor.php [hunk 2]
```diff
@@ -182,10 +176,17 @@ public function setElementsValue(string $search, array $elements, bool $inheritS
         ? $this->splitParagraphIntoParagraphs($block, $paragraphStyle, $textRunStyle)
         : $this->splitTextIntoTexts($block, $textRunStyle);
       if ($inheritStyle) {
-        $elementsData = str_replace(['<w:pPr/>', '<w:rPr/>'], [$paragraphStyle, $textRunStyle], $elementsData);
+        $elementsDataList = preg_replace_callback_array([
+          '#<w:pPr/>#' => fn() => $paragraphStyle,
+          '#<w:pPr.*</w:pPr>#' => fn (array $matches) => StyleMerger::mergeStyles($matches[0], $paragraphStyle),
+          // <w:pPr> may contain <w:rPr> itself so we have to math for <w:rPr> inside of <w:r>
```
Comment (jensschuppe) [inline CRM/Civioffice/DocumentRendererType/LocalUnoconv/PhpWordTemplateProcessor.php:182]: `s/math/match`

File: CRM/Civioffice/DocumentRendererType/LocalUnoconv/PhpWordTemplateProcessor.php [hunk 3]
```diff
@@ -212,8 +213,9 @@ public function splitParagraphIntoParagraphs(
     preg_match('#<w:pPr.*</w:pPr>#i', $paragraph, $matches);
     $extractedParagraphStyle = $matches[0] ?? '';
 
-    preg_match('#<w:rPr.*</w:rPr>#i', $paragraph, $matches);
-    $extractedTextRunStyle = $matches[0] ?? '';
+    // <w:pPr> may contain <w:rPr> itself so we have to math for <w:rPr> inside of <w:r>
```
Comment (jensschuppe) [inline CRM/Civioffice/DocumentRendererType/LocalUnoconv/PhpWordTemplateProcessor.php:216]: `s/math/match`

**Review by jensschuppe — CHANGES_REQUESTED** [review]

File: CRM/Civioffice/DocumentRendererType/LocalUnoconv/PhpWordTemplateProcessor.php (hunk 2 above)
Comment (dontub) [reply CRM/Civioffice/DocumentRendererType/LocalUnoconv/PhpWordTemplateProcessor.php]: Fixed.

File: CRM/Civioffice/DocumentRendererType/LocalUnoconv/PhpWordTemplateProcessor.php (hunk 3 above)
Comment (dontub) [reply CRM/Civioffice/DocumentRendererType/LocalUnoconv/PhpWordTemplateProcessor.php]: Fixed.

File: tests/phpunit/bootstrap.php (hunk 1 above)
Comment (dontub) [reply tests/phpunit/bootstrap.php]: Replaced by
```php
if (!function_exists('ts')) {
  // Ensure function ts() is available - it's declared in the same file as CRM_Core_I18n in CiviCRM < 5.74.
  // In later versions the function is registered following the composer conventions.
  \CRM_Core_I18n::singleton();
}
```

**Review by jensschuppe — APPROVED** [review]

**Pull request merged**